dataService.js

```diff
@@ -0,0 +1,94 @@
+// Maps
+let productMap;
+let categoryMap;
+let combinedProductMap;
+
+/*
+ * Method to create and return map of products if its not already created
+ */
+function getProducts() {
+
+  if(productMap) {
+    return productMap;
+  }
+
+  const productData = require('./src/data/products.json');
+
+  // Method variable to be stored in productMap
+  function createProductMap() {
+    const productMapToBeReturned = {};
+
+    productData.products.forEach(product => {
+      productMapToBeReturned[product.id] = product;
+    });
+    return productMapToBeReturned;
+  }
+
+  return productMap = createProductMap();
+}
+
+/*
+ * Method to create and return map of categories if not already created
+ */
+function getCategories() {
+
+  if (categoryMap) {
+    return categoryMap;
+  }
+
+  const categoryData = require('./src/data/categories.json');
+
+  // Method variable to be stored in categoryMap
+  function createCategoryMap() {
+    const categoryMapToBeReturned = {};
+
+    categoryData.categories.forEach(category => {
+      categoryMapToBeReturned[category.id] = category.categoryName;
+    });
+    return categoryMapToBeReturned;
+  }
+
+  return categoryMap = createCategoryMap();
+}
+
+/*
+ * Method to combine product and category data into combinedProductMap
+ */
+function combineProductsWithCategories() {
+
+  // Creating categoryMap and getting required data
+  getCategories();
+  const productData = require('./src/data/products.json');
+
+  combinedProductMap = [];
+
+  productData.products.forEach(product => {
+    // Making deep copy to prevent changes in actual data
+    let productToBeReturned = Object.assign({}, product);
+
+    // Adding categoryName
+    productToBeReturned.categoryName = categoryMap[productToBeReturned.categoryId];
+
+    combinedProductMap.push(productToBeReturned);
+  });
+}
+
+/*
+ * Creates combinedProductMap if its null
+ */
+function getCombinedProductMap() {
+
+  if (!combinedProductMap) {
+    combineProductsWithCategories();
+  }
+  return combinedProductMap;
+}
+
+module.exports = {
+  getProducts: getProducts(),
+  getCategories: getCategories(),
+  combineProductsWithCategories: combineProductsWithCategories(),
+  getCombinedProductMap: getCombinedProductMap(),
+  productMap: productMap,
+  categoryMap: categoryMap,
```

Comment:
Now that `getProducts()`, `getCategories()` and also `getCombinedProductMap()` are returning values, I think we no longer need to export `productMap`, `categoryMap` and `combineProductsWithCategories()` 

Waiting for mentors say on this @suthagar23 @f4ww4z

Reply:
Yes, Export only the required method. Don't export everything

Reply:
Done, Thanks!